File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/budgeting/TokenCostBudgeter.java
package com.cleverferret.v2.feature.ai.budgeting;

public final class TokenCostBudgeter {
    private final int maxPromptTokens;
    private final int maxCompletionTokens;
    private final double maxCostUsd;

    public TokenCostBudgeter(int maxPromptTokens, int maxCompletionTokens, double maxCostUsd) {
        this.maxPromptTokens = maxPromptTokens;
        this.maxCompletionTokens = maxCompletionTokens;
        this.maxCostUsd = maxCostUsd;
    }

    public BudgetDecision evaluate(int promptTokens, int completionTokens, double expectedCostUsd) {
        if (promptTokens > maxPromptTokens) {
            return BudgetDecision.rejected("prompt token budget exceeded");
        }
        if (completionTokens > maxCompletionTokens) {
            return BudgetDecision.rejected("completion token budget exceeded");
        }
        if (expectedCostUsd > maxCostUsd) {
            return BudgetDecision.rejected("cost budget exceeded");
        }
        return BudgetDecision.approved();
    }
}

File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/budgeting/TokenCostBudgeter.kt
package com.cleverferret.v2.feature.ai.budgeting

class TokenCostBudgeter(
    private val maxPromptTokens: Int,
    private val maxCompletionTokens: Int,
    private val maxCostUsd: Double,
) {
    fun evaluate(promptTokens: Int, completionTokens: Int, expectedCostUsd: Double): BudgetDecision {
        if (promptTokens > maxPromptTokens) return BudgetDecision.rejected("prompt token budget exceeded")
        if (completionTokens > maxCompletionTokens) return BudgetDecision.rejected("completion token budget exceeded")
        if (expectedCostUsd > maxCostUsd) return BudgetDecision.rejected("cost budget exceeded")
        return BudgetDecision.approved()
    }
}
